src/dht.hpp

```diff
@@ -0,0 +1,100 @@
+/**
+ * @file DHT.hpp
+ * @brief     DHT sensor class
+ * @author    Michel Baartman
+ * @license   See LICENSE
+ */
+
+#ifndef DHT_HPP
+#define DHT_HPP
+
+#include "wrap-hwlib.hpp"
+
+class DHT {
+  private:
+    hwlib::pin_in_out &pin; ///< DATA pin for DHT sensor
+    int temperature;        ///< Last environment temperature read
+    int version;            ///< dht sensor version
+    uint8_t data[5];        ///< data bytes sampled from sensor
+
+  public:
```

Comment:
Make functions that are only called by other functions of this class private. Ex: checksum, dhtReadWait and dhtReadGetBits (all called only by dhtRead)